pkgs/tools/security/nwaku/default.nix

```diff
@@ -0,0 +1,85 @@
+{
+  stdenv, fetchgit, lib, which, writeScriptBin,
+  # Options: wakunode1, wakunode2, wakubridge
+  makeTargets ? [ "wakunode2" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nwaku";
+  version = "0.11";
+  commit = "fec139748";
+  name = "${pname}-${version}-${commit}";
+
+  src = fetchgit {
+    url = "https://github.com/status-im/nwaku.git";
+    rev = "v${version}";
+    sha256 = "sha256-BAjueqcsoft8O5gP20jAKebZ1jj69VI79U+0U5vcsYk=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = let
+    # Fix for Nim compiler calling 'git rev-parse' and 'lsb_release'.
+    fakeGit = writeScriptBin "git" "echo $commit";
+    fakeLsbRelease = writeScriptBin "lsb_release" "echo nix";
+  in [ fakeGit fakeLsbRelease which ];
+
+  enableParallelBuilding = true;
+
+  # Avoid make calling 'git describe'.
+  GIT_VERSION = version;
+
+  NIMFLAGS = lib.optionalString (!nativeBuild) " -d:disableMarchNative";
+
+  makeFlags = makeTargets;
+
+  preBuildPhases = [ "buildCompiler" ];
+
+  # Generate vendor/.nimble contents with correct paths.
+  configurePhase = ''
+    export EXCLUDED_NIM_PACKAGES=""
+    export NIMBLE_LINK_SCRIPT=$PWD/vendor/nimbus-build-system/scripts/create_nimble_link.sh
+    export NIMBLE_DIR=$PWD/vendor/.nimble
+    export PWD_CMD=$(which pwd)
+    patchShebangs scripts > /dev/null
+    patchShebangs $PWD/vendor/nimbus-build-system/scripts > /dev/null
+    for dep_dir in $(find vendor -type d -maxdepth 1); do
+        pushd "$dep_dir" >/dev/null
+        $NIMBLE_LINK_SCRIPT "$dep_dir"
+        popd >/dev/null
+    done
+  '';
+
+  # Nimbus uses it's own specific Nim version bound as a Git submodule.
+  buildCompiler = ''
+    # Necessary for nim cache creation
+    export HOME=$PWD
+    make -j$NIX_BUILD_CORES build-nim
+    export PATH="$PWD/vendor/nimbus-build-system/vendor/Nim/bin:$PATH"
+  '';
+
+  installPhase = ''
```

Comment:
Should we add the pre/post hooks in the install phase? This can be important for users that call `overrideAttrs`.